test/ocl_ext_context.cpp

```diff
@@ -136,3 +139,123 @@ TEST(OCLCheck, DevicePlatform) {
 #else
 TEST(OCLExtContext, NoopCPU) {}
 #endif
+
+#pragma GCC diagnostic push
+#pragma GCC diagnostic ignored "-Wdeprecated-declarations"
```

Comment:
We have to figure out a generic way (a custom drop in marco perhaps) to do this for gcc, clang and msvc if possible. Instead of adding compiler specific pragmas in multiple locations.

Not as a part of this PR, of course.